torch/_meta_registrations.py

```diff
@@ -3609,7 +3609,7 @@ def meta__convert_weight_to_int4pack_for_cpu(w, inner_k_tiles):
 @register_meta([aten._weight_int4pack_mm])
 def meta__weight_int4pack_mm(x, w, q_group_size, q_scale_and_zeros):
     torch._check(x.dim() == 2, lambda: "x must be a 2D tensor")
-    torch._check(w.dim() == 4, lambda: "w must be a 4D tensor")
+    torch._check(w.dim() == 2 or w.dim() == 4, lambda: "w must be a 4D tensor")
```

Comment:
```suggestion
    torch._check(w.dim() == 2 or w.dim() == 4, lambda: "w must be a 2D tensor or 4D tensor")
```

Reply:
the change is moved to https://github.com/pytorch/pytorch/pull/149927. The comment is resolved.